Controller status push fails: missing voltage/current/position fields (422)

## Description
When changing train speed, the controller logs report the following error:

```
2025-12-10 13:19:00,838 [ERROR] app.mqtt_client: Failed to push status to central API: 422 {"detail":[{"type":"missing","loc":["body","voltage"],"msg":"Field required","input":null,"url":"https://errors.pydantic.dev/2.5/v/missing"},{"type":"missing","loc":["body","current"],"msg":"Field required","input":null,"url":"https://errors.pydantic.dev/2.5/v/missing"},{"type":"missing","loc":["body","position"],"msg":"Field required","input":null,"url":"https://errors.pydantic.dev/2.5/v/missing"}]}
```

## Steps to Reproduce

1. Use the UI or API to set speed on a locomotive
2. Locomotive does not move
3. Error provided is shown in controller logs

## Expected Behavior

The train should move as expected

## Additional Context

  ```
  2025-12-10 13:19:00,838 [ERROR] app.mqtt_client: Failed to push status to central API: 422 {"detail":[{"type":"missing","loc":["body","voltage"],"msg":"Field required","input":null,"url":"https://errors.pydantic.dev/2.5/v/missing"},{"type":"missing","loc":["body","current"],"msg":"Field required","input":null,"url":"https://errors.pydantic.dev/2.5/v/missing"},{"type":"missing","loc":["body","position"],"msg":"Field required","input":null,"url":"https://errors.pydantic.dev/2.5/v/missing"}]}
  ```

I believe this to be due to the controller images needing to be rebuilt on the RPi.

--- a/infra/ansible/inventory/production/hosts.yml
+++ b/infra/ansible/inventory/production/hosts.yml
@@ -88,6 +88,7 @@ all:
         # Set to 'true' to build image on RPi from source
         # Set to 'false' to pull pre-built image from GHCR
         build_edge_image: true
+        build_edge_nocache: false
 
         # Docker registry (same as central)
         docker_registry: ghcr.io

--- a/infra/ansible/playbooks/deploy_edge_multi_motor.yml
+++ b/infra/ansible/playbooks/deploy_edge_multi_motor.yml
@@ -77,9 +77,15 @@
     # Pull Latest Image (once for all motors)
     # ========================================================================
 
+
     - name: Check if we should build locally
       set_fact:
         should_build: "{{ build_edge_image | default(false) }}"
+
+    - name: Check if we should use build cache
+      set_fact:
+        build_nocache: "{{ build_edge_nocache | default(false) }}"
+      when: should_build | bool
 
     - name: Copy edge controller source to RPi (if building)
       synchronize:
@@ -101,6 +107,7 @@
         build:
           path: /tmp/edge-controller-build
           dockerfile: Dockerfile
+          nocache: "{{ build_nocache | bool }}"
         force_source: yes
       when: should_build | bool
       register: build_result

--- a/infra/docker/docker-compose.yml
+++ b/infra/docker/docker-compose.yml
@@ -59,21 +59,6 @@ services:
     stdin_open: true
     tty: true
 
-  edge-controller:
-    profiles: ["local-dev", "remote-dev", "prod"]
-    build:
-      context: ../../edge-controllers/pi-template
-      dockerfile: ${EDGE_CONTROLLER_DOCKERFILE:-Dockerfile.dev}
-    depends_on:
-      mqtt:
-        condition: service_started
-      central_api:
-        condition: service_healthy
-    environment:
-      - MQTT_BROKER=mqtt
-      - MQTT_PORT=1883
-      - LOCAL_DEV=${LOCAL_DEV:-false}
-
   debug:
     profiles: ["local-dev"]
     image: nicolaka/netshoot:latest

